File: lib/src/domain/entities/query.dart
import 'package:equatable/equatable.dart';

import 'package:query_wizard/domain.dart';

class Query extends Equatable implements IEntity {
  const Query({
    required this.id,
    required this.name,
    required this.sources,
    required this.tables,
    required this.fields,
    required this.joins,
    required this.groupings,
    required this.aggregates,
    required this.conditions,
    required this.orders,
    required this.isTop,
    required this.topCounter,
    required this.isDistinct,
  });

  Query.empty()
      : id = '',
        name = 'Query',
        sources = [],
        tables = [],
        fields = [],
        joins = [],
        groupings = [],
        aggregates = [],
        conditions = [],
        orders = [],
        isTop = false,
        topCounter = 0,
        isDistinct = false;

  @override
  final String id;
  final String name;
  final List<QueryElement> sources;
  final List<QueryElement> tables;
  final List<QueryElement> fields;
  final List<QueryJoin> joins;
  final List<QueryGrouping> groupings;
  final List<QueryAggregate> aggregates;
  final List<QueryCondition> conditions;
  final List<QueryOrder> orders;
  final bool isTop;
  final int topCounter;
  final bool isDistinct;

  @override
  List<Object?> get props => [
        id,
        name,
        sources,
        tables,
        fields,
        joins,
        groupings,
        aggregates,
        conditions,
        orders,
        isTop,
        topCounter,
        isDistinct,
      ];
}

File: server/src/QueryWizard.Domain/src/Entities/Query.cs
using System.Collections.Generic;

namespace QueryWizard.Domain.Entities
{
    public class Query
    {
        public string Id { get; set; }

        public string Name { get; set; }

        public List<QueryElement> Sources { get; set; }

        public List<QueryElement> Tables { get; set; }

        public List<QueryElement> Fields { get; set; }

        public List<QueryJoin> Joins { get; set; }

        public List<QueryGrouping> Groupings { get; set; }

        public List<QueryAggregate> Aggregates { get; set; }

        public List<QueryCondition> Conditions { get; set; }

        public List<QueryOrder> Orders { get; set; }

        public bool IsTop { get; set; }

        public int TopCounter { get; set; }

        public bool IsDistinct { get; set; }
    }
}
